spider_cms.py

```diff
@@ -643,16 +651,16 @@ def run(self):
                     return
                 continue
 
-            self.count_in += 1
-            self.buffer.append(next_doc)
+            self.count_in += len(next_batch)
+            self.buffer.extend(next_batch)
 
             if self.count_in%self.report_every == 0:
                 logging.debug("Processed %d docs" % self.count_in)
 
             # If buffer is full, send the docs and clear the buffer
-            if len(self.buffer) == self.bunch_size:
-                self.output_queue.put(self.buffer)
-                self.buffer = []
+            if len(self.buffer) >= self.bunch_size:
+                self.output_queue.put(self.buffer[:self.bunch_size])
```

Comment:
Why not just put the whole buffer?

Reply:
No strong reason, just that this way the bunches have always exactly the configured bunch size.